eclipse/src/saros/ui/wizards/pages/ResourceSelectionWizardPage.java

```diff
@@ -104,23 +106,24 @@ public void run() {
 
             final List<IResource> selectedResources = new ArrayList<IResource>();
 
-            if (preselectedResources != null && preselectedResources.size() > 0) {
+            resourceSelectionComposite.addResourceSelectionListener(resourceSelectionListener);
 
+            if (preselectedResources != null && preselectedResources.size() > 0) {
               selectedResources.addAll(preselectedResources);
               resourceSelectionComposite.setSelectedResources(selectedResources);
             }
 
             preselectedResources = null; // GC
 
-            resourceSelectionComposite.addResourceSelectionListener(resourceSelectionListener);
             /*
              * If nothing is selected and only one project exists in the
              * workspace, select it in the Wizard.
              */
-            if (selectedResources.size() == 0
-                && resourceSelectionComposite.getProjectsCount() == 1) {
+            List<IProject> displayedProjects = resourceSelectionComposite.getDisplayedProjects();
```

Comment:
Seems this is a relict where Saros could only share projects and therefore this behavior made sense. Any reason to not throw it away ?

Reply:
I was not sure about it. My guess was that sharing whole projects is the default use case so, if there is only a single project and we don't have any other pre-selection, it is a good guess. But I can remove it.

Reply:
This actually is hurtful to the logic as it is not checked correctly whether the project is actual a valid reference point. Removed the logic.

Reply:
Done.